search/query_params.hpp

```diff
@@ -27,28 +48,36 @@ struct QueryParams
   inline bool LastTokenIsPrefix() const { return !m_prefixTokens.empty(); }
 
   inline bool IsEmpty() const { return GetNumTokens() == 0; }
-  inline bool IsLangExist(int8_t lang) const { return m_langs.count(lang) != 0; }
-
   void Clear();
 
   bool IsCategorySynonym(size_t i) const;
+  TypeIndices & GetTypeIndices(size_t i);
+  TypeIndices const & GetTypeIndices(size_t i) const;
+
   bool IsPrefixToken(size_t i) const;
-  TSynonymsVector const & GetTokens(size_t i) const;
-  TSynonymsVector & GetTokens(size_t i);
+  Synonyms const & GetTokens(size_t i) const;
+  Synonyms & GetTokens(size_t i);
 
   // Returns true if all tokens in [start, end) range have integral
   // synonyms.
   bool IsNumberTokens(size_t start, size_t end) const;
 
   void RemoveToken(size_t i);
 
-  vector<TSynonymsVector> m_tokens;
-  TSynonymsVector m_prefixTokens;
-  vector<vector<uint32_t>> m_typeIndices;
+  inline Langs & GetLangs() { return m_langs; }
+  inline Langs const & GetLangs() const { return m_langs; }
+  inline bool IsLangExist(int8_t lang) const { return m_langs.count(lang) != 0; }
+
+  inline int GetScale() const { return m_scale; }
 
-  TLangsSet m_langs;
+private:
+  friend string DebugPrint(search::QueryParams const & params);
+
+  vector<Synonyms> m_tokens;
+  Synonyms m_prefixTokens;
+  vector<TypeIndices> m_typeIndices;
```

Comment:
Not now, as I said above, this change will be irrelevant to the logic of the PR. Let's do this in an individual PR, for the whole search subdirectory.